frontend/src/pages/main/profile/components/profileInformation/ProfileInformation.tsx

```diff
@@ -0,0 +1,92 @@
+import {UserModification} from "../../../../../types/user";
+import {useAppDispatch, useAppSelector} from "../../../../../stores/redux/hooks";
+import {logOut, selectLoggedUser} from "../../../../../stores/redux/loggedUser/loggedUserSlice";
+import {updateUserService} from "../../../../../services/user";
+import ConfirmationButton from "../../../../../components/buttons/confirmationButton/ConfirmationButton";
+import {Buttons} from "../../../../../common/constansts/buttons";
+import {FieldsValidation} from "../../../../auth/signup/types";
+import {showNotification} from "../../../../../stores/redux/notification/notificationSlice";
+import {NotificationType} from "../../../../../types/notification";
+import {Notifications} from "../../../../../common/constansts/notifications";
+import {Titles} from "../../../../../common/constansts/titles";
+import CardTitle from "../../../../../components/typography/cardTitle/CardTitle";
+
+const initialFieldsValidation: FieldsValidation = {
+    isEmailValid: false,
+    isPasswordValid: true,
+    isNameValid: false
+};
+
+const initialUserModification: UserModification = {
+    name: "",
+    email: "",
+    password: ""
+}
+
+const ProfileInformation = () => {
+
+    const [userModification, setUserModification] = useState<UserModification>(initialUserModification);
+    const [fieldsValidation, setFieldsValidation] = useState<FieldsValidation>(initialFieldsValidation);
+    const [isValid, setIsValid] = useState<boolean>(false);
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const updateEmail = (email: string, isEmailValid: boolean) => {
+        setUserModification({...userModification, email});
+        setFieldsValidation({...fieldsValidation, isEmailValid});
+    }
+
+    const updateName = (name: string, isNameValid: boolean) => {
+        setUserModification({...userModification, name});
+        setFieldsValidation({...fieldsValidation, isNameValid});
+    }
+
+    const updatePassword = (password: string, isPasswordValid: boolean) => {
+        setUserModification({...userModification, password});
+        if (password === "") {
+            const validateEmptyPassword = !isPasswordValid;
+            setFieldsValidation({...fieldsValidation, isPasswordValid: validateEmptyPassword});
+        } else {
+            setFieldsValidation({...fieldsValidation, isPasswordValid});
+        }
+    }
+
+    const validateData = () => {
+        const fieldsValid = Object.values(fieldsValidation).every(field => field);
+        setIsValid(fieldsValid);
+    };
+
+    const handleConfirm = async () => {
+        const updatedUser = await updateUserService(loggedUser.id, loggedUser.userType, userModification);
+        if (!updatedUser) return;
+        dispatch(showNotification({
+            show: true,
+            type: NotificationType.SUCCESS,
+            message: Notifications.SUCCESSFUL_UPDATE_USER_DATA
+        }));
+        dispatch(logOut());
```

Comment:
It's not user friendly. 
After update, you have to send a request to auth manager to generate new token instead of force log out.
But if you are lazy to make additional changes on the backend, just provide a clear message: Your data has been updated, please login again.